src/contrib/cluster/Akka.Cluster.Sharding.Tests/InactiveEntityPassivationSpec.cs

```diff
@@ -0,0 +1,149 @@
+        internal class Passivate
+        {
+            public static readonly Passivate Instance = new Passivate();
+            private Passivate() { }
+        }
+
+        internal class Entity : UntypedActor
+        {
+            private readonly string _id = Context.Self.Path.Name;
+
+            public IActorRef Probe { get; }
+
+            public static Props Props(IActorRef probe) =>
+                Actor.Props.Create(() => new Entity(probe));
+
+            public Entity(IActorRef probe)
+            {
+                Probe = probe;
+            }
+
+            protected override void OnReceive(object message)
+            {
+                switch (message)
+                {
+                    case Passivate _:
+                        Probe.Tell($"{_id} passivating");
+                        Context.Stop(Self);
+                        break;
+                    default:
+                        Probe.Tell(new GotIt(_id, message, DateTime.Now.Ticks));
+                        break;
+                }
+            }
+
+            public class GotIt
+            {
+                public string Id { get; }
+                public object Msg { get; }
+                public long When { get; }
+
+                public GotIt(string id, object msg, long when)
+                {
+                    Id = id;
+                    Msg = msg;
+                    When = when;
+                }
+            }
+        }
+
+        #endregion
+
+        private readonly ExtractEntityId _extractEntityId = message =>
+            message is int msg ? Tuple.Create(msg.ToString(), message) : null;
+
+        private readonly ExtractShardId _extractShard = message =>
+            message is int msg ? (msg % 10).ToString(CultureInfo.InvariantCulture) : null;
+
+        public InactiveEntityPassivationSpec()
+            : base(GetConfig())
+        { }
+
+        public static Config GetConfig()
+        {
+            return ConfigurationFactory.ParseString(@"
+                akka.loglevel = INFO
+                akka.actor.provider = cluster
+                akka.cluster.sharding.passivate-idle-entity-after = 3s")
+                .WithFallback(ClusterSharding.DefaultConfig())
+                .WithFallback(ClusterSingletonManager.DefaultConfig());
+        }
+
+        [Fact]
+        public void Passivation_of_inactive_entities_must_passivate_entities_when_they_have_not_seen_messages_for_the_configured_duration()
+        {
+            // Single node cluster
+            Cluster.Get(Sys).Join(Cluster.Get(Sys).SelfAddress);
+
+            var probe = new TestProbe(Sys, new XunitAssertions());
+            var settings = ClusterShardingSettings.Create(Sys);
+            var region = ClusterSharding.Get(Sys).Start(
+                "myType",
+                Entity.Props(probe.Ref),
+                settings,
+                _extractEntityId,
+                _extractShard,
+                new LeastShardAllocationStrategy(10, 3),
```

Comment:
`ClusterSharding.DefaultShardAllocationStrategy()` seems to be accessible from here in the JVM (see [#L69](https://github.com/johanandren/akka/blob/27296b4c808f967f6d636ef44c0a958fc93e0a44/akka-cluster-sharding/src/test/scala/akka/cluster/sharding/InactiveEntityPassivationSpec.scala#L69))